### ISSUE TITLE
Squeel and polyamorous dependencie problem

### ISSUE THREAD

**Issue by benoittgt** [activerecord-hackery/squeel#396]
Hello there

Thanks for the job done with squeel. I'm having an issue. I've wanted to update my project but had problem with `squeel` gem. Disable it in my gemfile, run a `bundle update` everything was ok except now when I turn it back I have an issue with polyarmous dependencie. I can't install it back. My only solution is to rollback and not update few gems. Any idea of what can be done ?

```
Bundler could not find compatible versions for gem "polyamorous":
  In snapshot (Gemfile.lock):
    polyamorous (= 1.2.0)

  In Gemfile:
    activeadmin (>= 0) ruby depends on
      ransack (~> 1.3) ruby depends on
        polyamorous (~> 1.2) ruby

    squeel (>= 0) ruby depends on
      polyamorous (~> 0.5.0) ruby

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
Bundler could not find compatible versions for gem "activerecord":
  In snapshot (Gemfile.lock):
    activerecord (= 4.2.3)

  In Gemfile:
    rails (= 4.2.3) ruby depends on
      activerecord (= 4.2.3) ruby

    activerecord-session_store (>= 0) ruby depends on
      activerecord (< 5, >= 4.0.0) ruby

    paper_trail (>= 0) ruby depends on
      activerecord (< 6.0, >= 3.0) ruby

    activeadmin (>= 0) ruby depends on
      ransack (~> 1.3) ruby depends on
        polyamorous (~> 1.2) ruby depends on
          activerecord (>= 3.0) ruby

    lol_dba (>= 0) ruby depends on
      activerecord (< 5.0, >= 3.0) ruby

    activeadmin (>= 0) ruby depends on
      ransack (~> 1.3) ruby depends on
        polyamorous (~> 1.2) ruby depends on
          activerecord (>= 3.0) ruby

    nilify_blanks (>= 0) ruby depends on
      activerecord (>= 3.0.0) ruby

    squeel (>= 0) ruby depends on
      activerecord (~> 3.1.0.alpha) ruby

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

**Comment by adamtao** [comment]
Looks like the same problem was reported in [issue 379](https://github.com/activerecord-hackery/squeel/issues/379)

A couple of solutions there. It looks like it has been fixed, but the gem hasn't been updated yet. So, I just changed my Gemfile to pull the latest from the github repository for now:

```ruby
gem 'squeel', github: 'activerecord-hackery/squeel'
```

**Comment by benoittgt** [comment]
Thanks @adamtao will try.

**Comment by jeffleeismyhero** [comment]
I had this issue only when including Ransack and Squeel with Rails 4.2.3. Using the git repo for Squeel as mentioned in @adamtao's comment worked for me.

**Comment by anhkind** [comment]
I'm using Rails 3.2.22 and it didn't seem to work with the latest gem version, i.e. 1.2.3, as it still has conflicts with activeadmin gem. @adamtao's comment didn't work for me. Can we update 'polyamorous' gem to version 1.2.x?

**Comment by Skulli** [comment]
Same problem here with rails **4.2.5** and latest *activeadmin* version, which depends on *polyamorous* **~>1.2**

**Comment by jfeldstein** [comment]
Worked for me w Rails 4.2.1 and activeadmin 1.0.0.pre2

**Comment by grimmwerks** [comment]
I'm having a hell of a time updating a project with active admin / squeel, etc.  This is finally where I end up:

```
Bundler could not find compatible versions for gem "polyamorous":
  In Gemfile:
    polyamorous

    ransack was resolved to 1.8.1, which depends on
      polyamorous (~> 1.3)

    squeel was resolved to 1.2.3, which depends on
      polyamorous (~> 1.1.0)
```
Have been trying with different versions of active admin:

```
# gem 'activeadmin', '~> 1.0.0.pre2'
# gem 'activeadmin', github: 'activeadmin'
gem 'activeadmin', github: 'gregbell/active_admin'
```
and squeel:
```
# gem 'squeel', '~> 1.2', '>= 1.2.3'
# gem "squeel"  # Last officially released gem
gem 'squeel', github: 'activerecord-hackery/squeel'
# gem "squeel", :git => "git://github.com/ernie/squeel.git" # Track git repo
```

If I comment out squeel and bundle update it's no problem.

**Comment by benoittgt** [comment]
Stale issue. Closing for now.
